clang/lib/CodeGen/CGHLSLRuntime.cpp

```diff
@@ -115,6 +116,18 @@ GlobalVariable *replaceBuffer(CGHLSLRuntime::Buffer &Buf) {
 
 } // namespace
 
+llvm::Type *CGHLSLRuntime::convertHLSLSpecificType(const Type *T) {
+  assert(T->isHLSLSpecificType() && "Not an HLSL specific type!");
+
+  // Check if the target has a specific translation for this type first.
+  if (llvm::Type *TargetTy = CGM.getTargetCodeGenInfo().getHLSLType(CGM, T))
+    return TargetTy;
+
+  // TODO: What do we actually want to do generically here? OpenCL uses a
+  // pointer in a particular address space.
+  llvm_unreachable("Generic handling of HLSL types is not implemented yet");
```

Comment:
At what point will this become something we need to do?  Is there an issue tracking it?

Reply:
I am not sure if we have any HLSL specific type that will not be handled by `getHLSLType`, so this might just become simple `llvm_unreachable` without the TODO: comment. @bogner?

Reply:
For DirectX and SPIR-V we'll implement `getHLSLType` and this should really be unreachable, but we may want to do something generic in the future if/when we want to generate code for GPU or CPU targets directly generically. Something like the OpenCL approach mentioned in the comment might make sense in that case.

Reply:
I have removed the comment and left the `llvm_unreachable`. We can revisit this in the future.